source/news/2021-04-22.html.markdown

```diff
@@ -0,0 +1,73 @@
+---
+title: dry-system 0.19 released with Zeitwerk support, plus component-handling flexibility and consistency improvements
+date: 2021-04-22
+author: timriley
+---
+
+We are very pleased to announce the release of dry-system 0.19.0! This release marks a huge step forward for dry-system, bringing support for [Zeitwerk][zw] and other autoloaders, plus clearer configuration and improved consistency around component resolution for both finalized and lazy loading containers.
+
+A dry-system container as of 0.19.0, configured for Zeitwerk, now looks like this:
+
+```ruby
+require "dry/system/container"
+require "dry/system/loader/autoloading"
+require "zeitwerk"
+
+class MyApp::Container < Dry::System::Container
+  configure do |config|
+    config.root = __dir__
+
+    config.component_dirs.loader = Dry::System::Loader::Autoloading
+    config.component_dirs.add_to_load_path = false
+
+    config.component_dirs.add "lib"
+  end
+end
+
+loader = Zeitwerk::Loader.new
+loader.push_dir MyApp::Container.config.root.join("lib").realpath
+loader.setup
+```
+
+Unlike the default loader, the new `Dry::System::Loader::Autoloading` does not `require` files itself when loading components. Instead, it references their class constants directly, allowing the missing constant resolution to trigger the autoloading behaviour of [Zeitwerk][zw] and other autoloaders. This is all that's required to bring Zeitwerk and dry-system together!
```

Comment:
@timriley I'd mention briefly here that even though constants can be autoloaded, it's still recommended that you're going to rely on auto-injection setup rather than creating tight coupling between components through class/module constants